internal/provider/resource_cluster.go

```diff
@@ -625,36 +674,34 @@ func resourceClusterUpdate(ctx context.Context, d *schema.ResourceData, meta int
 }
 
 // upgradeClusterVersion updates a cluster's Consul version to a valid upgrade version
-func upgradeClusterVersion(ctx context.Context, d *schema.ResourceData, meta interface{}, managedApp managedapplications.Application) diag.Diagnostics {
-	// Retrieve the valid upgrade versions
-	upgradeVersionsResponse, err := meta.(*clients.Client).CustomResourceProvider.ListUpgradeVersions(ctx, *managedApp.ManagedResourceGroupID)
-	if err != nil {
-		return diag.Errorf("unable to retrieve upgrade versions for HCS cluster (Managed Application ID %q) (Correlation ID %q): %v",
-			*managedApp.ID,
-			meta.(*clients.Client).CorrelationRequestID,
-			err,
-		)
-	}
+func upgradeCluster(ctx context.Context, meta interface{}, managedApp managedapplications.Application, update *models.HashicorpCloudConsulamaAmaClusterUpdate) diag.Diagnostics {
+	if update.ConsulVersion != "" {
+		// Retrieve the valid upgrade versions
+		upgradeVersionsResponse, err := meta.(*clients.Client).CustomResourceProvider.ListUpgradeVersions(ctx, *managedApp.ManagedResourceGroupID)
+		if err != nil {
+			return diag.Errorf("unable to retrieve upgrade versions for HCS cluster (Managed Application ID %q) (Correlation ID %q): %v",
+				*managedApp.ID,
+				meta.(*clients.Client).CorrelationRequestID,
+				err,
+			)
+		}
 
-	v, ok := d.GetOk("min_consul_version")
-	if !ok {
-		return diag.Errorf("min_consul_version is required in order to upgrade the cluster")
-	}
-	newConsulVersion := consul.NormalizeVersion(v.(string))
+		newConsulVersion := consul.NormalizeVersion(update.ConsulVersion)
 
-	if upgradeVersionsResponse.Versions == nil {
-		return diag.Errorf("no upgrade versions of Consul are available for this cluster; you may already be on the latest Consul version supported by HCS")
-	}
+		if upgradeVersionsResponse.Versions == nil {
+			return diag.Errorf("no upgrade versions of Consul are available for this cluster; you may already be on the latest Consul version supported by HCS")
+		}
 
-	if !consul.IsValidVersion(newConsulVersion, consul.FromAMAVersions(upgradeVersionsResponse.Versions)) {
-		return diag.Errorf("specified Consul version (%s) is unavailable; must be one of: %+v", newConsulVersion, consul.FromAMAVersions(upgradeVersionsResponse.Versions))
+		if !consul.IsValidVersion(newConsulVersion, consul.FromAMAVersions(upgradeVersionsResponse.Versions)) {
+			return diag.Errorf("specified Consul version (%s) is unavailable; must be one of: %+v", newConsulVersion, consul.FromAMAVersions(upgradeVersionsResponse.Versions))
+		}
 	}
```

Comment:
```suggestion
                update.ConsulVersion = newConsulVersion
	}
```
I can't be sure of the indentation on this suggestion but I think we need to add it.